google-cloud-storage/src/main/java/com/google/cloud/storage/MultipartUploadClientImpl.java

```diff
@@ -75,4 +77,12 @@ public AbortMultipartUploadResponse abortMultipartUpload(AbortMultipartUploadReq
         () -> httpRequestManager.sendAbortMultipartUploadRequest(uri, request),
         Decoder.identity());
   }
+
+  @Override
+  public UploadPartResponse uploadPart(UploadPartRequest request, RequestBody requestBody) {
+    return retrier.run(
+        retryAlgorithmManager.idempotent(),
+        () -> httpRequestManager.sendUploadPartRequest(uri, request, requestBody),
```

Comment:
This needs to do dirty state tracking, and to actually rewind the content before a followup invocation. See https://github.com/googleapis/java-storage/blob/763be91bd8532b4fc8c3a3c2248509c0c5256086/google-cloud-storage/src/main/java/com/google/cloud/storage/JsonResumableSession.java#L62-L76 for how this is done for json resumable uploads.

Additionally, this should pass the RewindableContent to `sendUploadPartRequest` so that this method manages the lifecycle of it and integrates correctly with the retries.